### ISSUE TITLE
Remove OnConnectionClosed and Abort from IConnectionContext

### ISSUE THREAD

**Issue by davidfowl** [aspnet/KestrelHttpServer#1772]
Once this is merged https://github.com/dotnet/corefxlab/pull/1491, we should be able to remove all use the pipe as a single communication mechanism for transports.

**Comment by muratg** [comment]
This is not part of the public API surface, so moving this out to 2.1.0.

**Comment by davidfowl** [comment]
Fixed in https://github.com/aspnet/KestrelHttpServer/commit/7854c0604a3e6e50a0bea84c4fa9d9e6ca5d7057
